File: fizzbuzz.exs
Enum.each 1..100, fn i ->
  case {rem(i, 3) == 0, rem(i, 5) == 0} do
    {true, true} ->
      IO.puts("Fizzbuzz")
    {true, false} ->
      IO.puts("Fizz")
    {false, true} ->
      IO.puts("Buzz")
    {false, false} ->
      IO.puts(i)
  end
end

File: fizzbuzz.erl
-module(fizzbuzz).
-export([execute/0]).

fizzbuzz(N) when N rem 15 == 0 -> "FizzBuzz";
fizzbuzz(N) when N rem 3  == 0 -> "Fizz";
fizzbuzz(N) when N rem 5  == 0 -> "Buzz";
fizzbuzz(N) -> integer_to_list(N).

execute() -> lists:map(fun fizzbuzz/1 , lists:seq(1, 100)).
